### PR TITLE
[FIX] l10n_be_intrastat: adapt to new regulation for belgian intrastat

### PR DESCRIPTION
**Pull request by qdp-odoo**
This is essentially a backport of module l10n_be_intrastat_2019 made for enterprise v12 + the refactoring needed on the original v10 module to allow the changes to apply where needed.

Was requested by OPW #1887019

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_be_intrastat_2019/models/account_intrastat_report.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    def _build_intrastat_line(self, numlgn, item, linekey, amounts, dispatchmode, extendedmode):
+        super(XmlDeclaration, self)._build_intrastat_line(numlgn, item, linekey, amounts, dispatchmode, extendedmode)
+        if dispatchmode:
+            self._set_Dim(item, 'EXCNTORI', unicode(linekey.EXCNTORI))
+            self._set_Dim(item, 'PARTNERID', unicode(linekey.PARTNERID))
+
+    def _get_intrastat_linekey(self, declcode, inv_line, dispatchmode, extendedmode):
+        res = super(XmlDeclaration, self)._get_intrastat_linekey(declcode, inv_line, dispatchmode, extendedmode)
+        if res and dispatchmode:
+            excntori = inv_line.intrastat_product_origin_country_id.code or 'QU'
```
[the remark this answers is not in the document]
Comment (qdp-odoo) [reply addons/l10n_be_intrastat_2019/models/account_intrastat_report.py]: it's up to you to change: I've done my part providing this and I let you finish the job :-)

**Comment by nim-odoo** [comment]
robodoo retry

**Comment by nim-odoo** [comment]
robodoo retry

**Comment by nim-odoo** [comment]
robodoo retry

**Pull request closed without merging**